changeloggenutils/minor_release.go

```diff
@@ -21,46 +22,58 @@ type MinorReleaseGroupedChangelogGenerator struct {
 }
 
 /**
- Groups releases by their minor version:
- v1.8
- - v1.8.2
- - v1.8.1
- v1.7
- - v1.7.1
- - v1.7.0-beta9
- ...
- */
+Groups releases by their minor version:
+v1.8
+- v1.8.2
+- v1.8.1
+v1.7
+- v1.7.1
+- v1.7.0-beta9
+...
+*/
 func NewMinorReleaseGroupedChangelogGenerator(opts Options, client *github.Client) *MinorReleaseGroupedChangelogGenerator {
-	if opts.NumVersions == 0{
+	if opts.NumVersions == 0 {
 		opts.NumVersions = math.MaxInt64
 	}
 	return &MinorReleaseGroupedChangelogGenerator{
-		opts:      opts,
-		Client:    client,
+		opts:   opts,
+		Client: client,
 	}
 }
 
 // Entry point for generating changelog JSON
 func (g *MinorReleaseGroupedChangelogGenerator) GenerateJSON(ctx context.Context) (string, error) {
 	var err error
-	releaseData, err := g.GetReleaseData(ctx)
+	releaseData, err := g.GetReleaseData(ctx, g.opts.MainRepoReleases)
 	if err != nil {
 		return "", err
 	}
 	var out struct {
 		Opts        Options
 		ReleaseData *ReleaseData
 	}
-	out.Opts = g.opts
+	out.Opts = Options{
+		RepoOwner:     g.opts.RepoOwner,
+		MainRepo:      g.opts.MainRepo,
+		DependentRepo: g.opts.DependentRepo,
+	}
 	out.ReleaseData = releaseData
 	res, err := json.Marshal(out)
 	return string(res), err
 }
 
-func (g *MinorReleaseGroupedChangelogGenerator) GetReleaseData(ctx context.Context) (*ReleaseData, error) {
-	releases, err := githubutils.GetAllRepoReleasesWithMax(ctx, g.Client, g.opts.RepoOwner, g.opts.MainRepo, g.opts.NumVersions)
-	if err != nil {
-		return nil, err
+func (g *MinorReleaseGroupedChangelogGenerator) GetReleaseData(ctx context.Context, cachedReleases []*github.RepositoryRelease) (*ReleaseData, error) {
```

Comment:
Updated this function to return `cachedReleases` if it exists, else to fetch the releases itself.